### DISCUSSION TITLE
Examples for all backend libraries

### DISCUSSION THREAD

**Discussion by s4093120** [rust-skia/rust-skia#688] [General]
So I was wondering if the examples could be updated for each backend library (eg. vulkan, directx and whatnot). Currently there is only metal and opengl examples which is nice. I'm personally trying to use vulkan using ash with skia.

**Comment by pragmatrix** [comment]
I would rather prefer separate backend projects with example code that are maintained by stakeholders outside of the rust-skia primary repository.

The OpenGL and Metal window examples are in this repository, because they are simple. Vulkan and DirectX are entirely different beasts as far as my experience goes.

We can add links to the README.md and move them into the rust-skia organization if maintainer responsibilities are clearly defined.

And for using ash, you might want to take a look at [Skulpin](https://github.com/aclysma/skulpin).

Moving this over to discussions.
